core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1543,10 +1535,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
     preferredReplicaImbalanceCount.set(pric)
   }
 
-  private def triggerControllerMove(): Unit = {
-    activeControllerId.set(-1)
```

Comment:
I'm not sure if removing this line is the right move, as it will cause events to be processed by the controller even after triggering the controller move.

`triggerControllerMove` now just deletes the `/controller` znode. The `Reelect` corresponding to the `/controller` znode deletion will come behind events already in the queue. Since the PR removes the `activeControllerId.set(-1)` line, the controller will still think it's active and will process the queued events in between the current event and the Reelect event.

When I first saw the PR title, I thought you'd simply add a line calling `onControllerResignation` within `triggerControllerMove`. I think this makes more sense.